alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -62,14 +59,23 @@ abstract class MonitorRunner {
                 throw IllegalStateException("Message content missing in the Destination with id: ${action.destinationId}")
             }
             if (!dryrun) {
-                // TODO: Inject user here so only Destination/Notifications that the user has permissions to are retrieved
-                withContext(Dispatchers.IO) {
-                    actionOutput[Action.MESSAGE_ID] = getConfigAndSendNotification(
-                        action,
-                        monitorCtx,
-                        actionOutput[Action.SUBJECT],
-                        actionOutput[Action.MESSAGE]!!
+                // TODO: Add integration test to ensure user context information is passed correctly when calling Notification plugin and
+                // only accessing notification channels that the user can only access
+                val userStr = monitorCtx.client!!.threadPool().threadContext
+                    .getTransient<String>(ConfigConstants.OPENSEARCH_SECURITY_USER_INFO_THREAD_CONTEXT)
+                monitorCtx.client!!.threadPool().threadContext.stashContext().use {
+                    monitorCtx.client!!.threadPool().threadContext.putTransient(
+                        ConfigConstants.OPENSEARCH_SECURITY_USER_INFO_THREAD_CONTEXT,
+                        userStr
                     )
+                    withContext(Dispatchers.IO) {
```

Comment:
I was wondering if we could just use `withClosableContext` here as well to pass in the roles of the user but I don't know if not using `Dispatchers.IO` has any side effects on coroutine behavior. We could maybe make a note of it and just proceed as-is for now and find out later.